SWLOR.Game.Server/Perk/DarkSide/ForceLeech.cs

```diff
@@ -0,0 +1,175 @@
+            _skill = skill;
+        }
+
+        public bool CanCastSpell(NWPlayer oPC, NWObject oTarget)
+        {
+            if (_.GetDistanceBetween(oPC, oTarget) > 15.0f)
+                return false;
+
+            return true;
+        }
+
+        public string CannotCastSpellMessage(NWPlayer oPC, NWObject oTarget)
+        {
+            return "Target out of range.";
+        }
+
+
+        public int FPCost(NWPlayer oPC, int baseFPCost)
+        {
+            return baseFPCost;
+        }
+
+        public float CastingTime(NWPlayer oPC, float baseCastingTime)
+        {
+            return baseCastingTime;
+        }
+
+        public float CooldownTime(NWPlayer oPC, float baseCooldownTime)
+        {
+            return baseCooldownTime;
+        }
+
+        public void OnImpact(NWPlayer player, NWObject target, int level)
+        {
+            float recoveryPercent;
+            var effectiveStats = _stat.GetPlayerItemEffectiveStats(player);
+            int darkBonus = effectiveStats.DarkAbility;
+            int length;
+            int dotAmount;
+            int wisdom = player.WisdomModifier;
+            int intelligence = player.IntelligenceModifier;
+
+
+            switch (level)
+            {
+                case 1:
+                    length = 6;
+                    dotAmount = 2;
+                    recoveryPercent = 0.5f;
+                    break;
+                case 2:
+                    length = 9;
+                    dotAmount = 2;
+                    recoveryPercent = 0.5f;
+                    break;
+                case 3:
+                    length = 6;
+                    dotAmount = 4;
+                    recoveryPercent = 0.5f;
+                    break;
+                case 4:
+                    length = 6;
+                    dotAmount = 4;
+                    recoveryPercent = 0.75f;
+                    break;
+                case 5:
+                    length = 9;
+                    dotAmount = 4;
+                    recoveryPercent = 0.75f;
+                    break;
+                case 6:
+                    length = 9;
+                    dotAmount = 6;
+                    recoveryPercent = 0.75f;
+                    break;
+                case 7:
+                    length = 12;
+                    dotAmount = 8;
+                    recoveryPercent = 0.5f;
+                    break;
+                case 8:
+                    length = 12;
+                    dotAmount = 8;
+                    recoveryPercent = 0.75f;
+                    break;
+                case 9:
+                    length = 15;
+                    dotAmount = 12;
+                    recoveryPercent = 0.5f;
+                    break;
+                case 10:
+                    length = 15;
+                    dotAmount = 12;
+                    recoveryPercent = 0.75f;
+                    break;
+                case 11: // Only available with background bonus
```

Comment:
This is more of a note to myself, but since I removed force backgrounds this case isn't necessary. Doesn't hurt anything to leave it in though.